### PR TITLE
Updated phrasing of belongsTo, hasMany and belongsToMany assocations

### PR DESCRIPTION
**Pull request by othercorey**
This follows the changes made to hasOne in https://github.com/cakephp/docs/pull/6209.

I stopped short of trying to adjust everything, this is just a high-level pass to sync up some of the options.

### CODE REVIEW

**Comment by othercorey** [comment]
I've never been completely sure if the table examples were supposed to show 'Entity hasMany Entity' or 'Table hasMany Table'.

I synced up the naming as most were using 'Table hasMany Table' but perhaps this could be improved.

There are several examples of record, Table, table, result, entity, etc all being mixed around with various spellings of Table(s), table(s), entity(s), Entity(s).

**Comment by markstory** [comment]
> I've never been completely sure if the table examples were supposed to show 'Entity hasMany Entity' or 'Table hasMany Table'.

'Table hasMany Table' is more inline with what the user will have to configure in the framework but it ends up reading a bit clunky in sentences which is likely why there is some inconsistencies.

File: en/orm/associations.rst [hunk 1]
```diff
@@ -356,11 +356,11 @@ convention:
 ========================== ===================
 Relation                   Schema
 ========================== ===================
-Article hasMany Comment    Comment.article\_id
+Articles hasMany Comments  Comments.article\_id
 -------------------------- -------------------
```
Comment (markstory) [inline en/orm/associations.rst]: You might have to extend the table underlines here. Sphinx is particular about lines matching their contents.

Comment (othercorey) [reply en/orm/associations.rst]: Always miss something with those.

**Pull request merged**